pkg/sharings/sharings.go

```diff
@@ -204,8 +202,39 @@ func ShareDoc(instance *instance.Instance, sharing *Sharing, recStatus *Recipien
 			}
 		}
 
+		var values []string
+
+		// Dynamic sharing
+		if rule.Selector != "" {
+			// Create index based on selector to retrieve documents to share
+			index := mango.IndexOnFields(docType, sharing.ID(), []string{rule.Selector})
+			err := couchdb.DefineIndex(instance, index)
+			if err != nil {
+				return err
+			}
+
+			var docs []couchdb.JSONDoc
+
+			// Request the index for all values
+			for _, val := range rule.Values {
```

Comment:
You can also use mango.Or(mango.Equal(), mango.Equal(), ...) to combine all values into a single a couchdb request.